roiextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -236,11 +240,18 @@ def __init__(
     def __del__(self):
         self.io.close()
 
-    def time_to_frame(self, time: FloatType):
-        return int((time - self._imaging_start_time) * self.get_sampling_frequency())
-
-    def frame_to_time(self, frame: IntType):
-        return float(frame / self.get_sampling_frequency() + self._imaging_start_time)
+    def time_to_frame(self, times: Union[FloatType, NumpyArray]):
+        if self._times is None:
+            return int((times - self._imaging_start_time) * self.get_sampling_frequency())
+        else:
+            return super().time_to_frame(times)
+            
+    def frame_to_time(self, frames: Union[IntType, NumpyArray]):
+        if self._times is None:
+            return float(frames / self.get_sampling_frequency() + self._imaging_start_time)
+        else:
+            return super().frame_to_time(frames)
```

Comment:
Wouldn't it make more sense to move this to the base class?

Reply:
No because `imaging_start_time` currently is only supported by NWB